Library/Homebrew/utils/analytics.rb

```diff
@@ -239,11 +292,17 @@ def clear_os_arch_prefix_ci
 
       def os_arch_prefix_ci
         @os_arch_prefix_ci ||= begin
-          os = OS_VERSION
-          arch = ", #{arch_label}" if arch_label.present?
-          prefix = ", #{custom_prefix_label}" unless Homebrew.default_prefix?
-          ci = ", CI" if ENV["CI"]
-          "#{os}#{arch}#{prefix}#{ci}"
+          data = {
+            os:        OS_VERSION,
+            developer: Homebrew::EnvConfig.developer?,
+            version:   HOMEBREW_VERSION,
+            system:    HOMEBREW_SYSTEM,
+            ci:        ENV["CI"].present?,
+          }
+          data[:arch] = arch_label if arch_label.present?
+          data[:prefix] = custom_prefix_label unless Homebrew.default_prefix?
```

Comment:
```suggestion
          data = {
            os:        OS_VERSION,
            developer: Homebrew::EnvConfig.developer?,
            version:   HOMEBREW_VERSION,
            system:    HOMEBREW_SYSTEM,
            ci:        ENV["CI"].present?,
            arch:      arch_label(verbose: verbose),
            prefix:    custom_prefix_label(verbose: verbose),
          }
          unless verbose
            data.delete(:arch) if data[:arch].blank?
            data.delete(:prefix) if Homebrew.default_prefix?
          end
```

Will also need modification to `arch_label` and `custom_prefix_label` have a `verbose` parameter that means they return a value for the default cases e.g. Intel and default prefixes.